web/src/compose_validate.js

```diff
@@ -681,9 +681,10 @@ function validate_private_message() {
     if (
         page_params.realm_private_message_policy ===
             settings_config.private_message_policy_values.disabled.code &&
-        (user_ids.length !== 1 || !people.get_by_user_id(user_ids[0]).is_bot)
+        (user_ids.length !== 1 ||
+            !(people.get_by_user_id(user_ids[0]).is_bot || people.is_my_user_id(user_ids[0])))
     ) {
```

Comment:
Could this check call `user_can_direct_message` to avoid duplicating that logic?

Doing so would also let us delete the "unless ..." comment, since the code reads clearly without it.

Reply:
Instead of duplicating the logic now it uses `user_can_direct_message` function.

Reply:
I don't understand. `user_can_direct_message` checks the policy, the recipient length, etc.; shouldn't a lot more of this code be replaced by this?

It feels like you're rushing here; please take the time to think through this whole PR clearly and do a self-review on it, and let me know when you feel it's 100% clean and correct.

Reply:
@sayyedarib this feedback is still pending.